### PR TITLE
Add seperate args for specifying unique uuid for boot root

### PR DESCRIPTION
**Pull request by jaskaran-khurana**

### CODE REVIEW

**Comment by jaskaran-khurana** [comment]
@chrisccoulson could you take a look please?

File: encrypt.go [hunk 1]
```diff
@@ -369,10 +372,11 @@ func (e *imageEncrypter) encryptImageOnDevice() error {
 
 	log.Infoln("encrypting image on", e.devPath)
 
-	if err := e.detectPartitions(); err != nil {
-		return err
+	if err := e.detectPartitions(e.opts.RootPartitionUUID, e.opts.EfiPartitionUUID); err != nil {
+			return err
```
Comment (canfikret) [inline encrypt.go:376]: Nit: remove extra tab.

File: deploy.go [hunk 2]
```diff
@@ -467,6 +468,24 @@ func (d *imageDeployer) run(opts *deployOptions) error {
 		return xerrors.Errorf("cannot obtain source file information: %w", err)
 	}
 
+   	rootPartitionUniqueUUID := false
+        efiPartitionUniqueUUID := false
+        if (opts.RootPartitionUUID != "") {
+                rootPartitionUniqueUUID = true
```
Comment (canfikret) [inline deploy.go:474]: Do you want to validate if input is correct UUID format?

File: main.go [hunk 3]
```diff
@@ -215,27 +215,44 @@ func (b *encryptCloudImageBase) connectNbd(path string) error {
...
 
-	// XXX: Could there be more than one partition with this type?
-	root := partitions.FindByPartitionType(linuxFilesystemGUID)
-	if root == nil {
-		return fmt.Errorf("cannot find partition with the type %v on %s", linuxFilesystemGUID, b.devPath)
+	var root *gpt.PartitionEntry = nil
+	if (rootPartitionUUID != "") {
+		root = partitions.FindByPartitionUniqueUUID(rootPartitionUUID)
+		if root == nil {
+			return fmt.Errorf("cannot find partition by partition name %s", rootPartitionUUID)
```
Comment (canfikret) [inline main.go]: Nit: by partition name -> by partition id

Comment (canfikret) [reply main.go]: Pls confirm: is it partition UUID or partition-type UUID that is in this context?

**Review by canfikret — CHANGES_REQUESTED** [review]

File: deploy.go (hunk 2 above)
Comment (jaskaran-khurana) [reply deploy.go:474]: yes we can but I do call string to uuid and that is called fairly early ...

File: main.go (hunk 3 above)
Comment (jaskaran-khurana) [reply main.go]: fixed

Comment (jaskaran-khurana) [reply main.go]: I am using the unique partition id which should be present in the gpt table and unique for each partition when I am explicitly specifying.

File: encrypt.go (hunk 1 above)
Comment (jaskaran-khurana) [reply encrypt.go:376]: acknowledging

**Pull request closed without merging**